gwu-libraries/scholarspace-hyrax
# T557 seeding dev data

Will close #557 

This PR: 

- Adds `factory` files from the [Hyrax repository](https://github.com/samvera/hyrax/tree/main/lib/hyrax/specs/shared_specs/factories) for use in testing and seeding of development data. 
- Adds `fixture` files from the Hyrax repository that are used in seeding works. 
- Updates several tests to work with the fixtures and factories taken from Hyrax. 
- Updates the `seeds.rb` file to:
  - Not work in `production`, should only be used for `development`. Use the `prep_new_prod` rake task if you need to set up a new production environment, which does not seed with demo works. 
  - Create the default admin set and an "ETDs" admin set.
  - Adds five basic collections.
  - Adds works with specific file/nesting conditions, though they are private and you must log in as the admin user to see them (password for FactoryBot generated users is `password`). 
  - Adds five public `GwWorks` with metadata filled in, should be viewable without logging in, and adds them to `Collection Title 1`. 
  - Adds five public `GwJournalIssues`
  - Adds five public `GwEtds`

There might (probably) be a cleaner way of importing the factories and fixtures from Hyrax, but the location of these files in their repository has changed recently, and we are on a different version than their most up-to-date - so simplest/most stable solution right now was to just directly copy from their repository and make tweaks where needed. Definitely open to feedback or other ideas for handling this. 

To test:
- In your development environment, drop, recreate, and migrate the development database (`rails db:drop` followed by `rails db:create` followed by `rails db:migrate`). 
- Run `rails db:seed`, which may take 2-3 minutes to run. Ensure there are no errors. 
- Start development server `rails s` and ensure that the created works are visible, and you are able to log in as any of the users created in the seeds file.
